src/mbgl/geometry/feature_index.cpp

```diff
@@ -23,17 +23,20 @@ FeatureIndex::FeatureIndex(std::unique_ptr<const GeometryTileData> tileData_)
 
 void FeatureIndex::insert(const GeometryCollection& geometries,
                           std::size_t index,
+                          FeatureType featureType,
                           const std::string& sourceLayerName,
                           const std::string& bucketLeaderID) {
-    auto featureSortIndex = sortIndex++;
     for (const auto& ring : geometries) {
         auto envelope = mapbox::geometry::envelope(ring);
         if (envelope.min.x < util::EXTENT &&
             envelope.min.y < util::EXTENT &&
             envelope.max.x >= 0 &&
             envelope.max.y >= 0) {
-            grid.insert(IndexedSubfeature(index, sourceLayerName, bucketLeaderID, featureSortIndex++),
+            grid.insert(IndexedSubfeature(index, sourceLayerName, bucketLeaderID, sortIndex++),
```

Comment:
Why it is important to change approach (featureSortIndex -> sortIndex change) for other than polygon features - was the removed code https://github.com/mapbox/mapbox-gl-native/pull/14884/commits/04012e422be622e764737f71c891f0c97a5a6da1#diff-c7acc1b0aee2a0628aac4613e46fa727L74 used only for polygons?

Reply:
AFAIK the logic for keeping the same sorting index makes sense for polygon features only, @ansis could you confirm on that?